apps/sms/test/marionette/mocks/mock_test_storages.js

```diff
@@ -0,0 +1,53 @@
+/* global Components, Services, PromiseUtils */
+'use strict';
+
+const Cu = Components.utils;
+Cu.import('resource://gre/modules/Services.jsm');
+Cu.import('resource://gre/modules/PromiseUtils.jsm');
+
+Services.obs.addObserver(function(document) {
+  if (!document || !document.location) {
+    return;
+  }
+
+  var window = document.defaultView;
+  var storages = new Map();
+
+  function getStorage(key) {
+    var storage = storages.get(key);
+
+    if (!storage) {
+      storage = {
+        deferred: PromiseUtils.defer(),
+        data: null
+      };
+
+      storages.set(key, storage);
+    }
+
+    return storage;
+  }
+
+
+  var TestStorages = {
+    getStorage: function(key) {
+      return getStorage(key).deferred.promise;
+    },
+
+    setStorage: function(key, value) {
+      getStorage(key).data = value;
+    },
+
+    setStorageReady: function(key) {
+      var storage = getStorage(key);
+
+      storage.deferred.resolve(Cu.waiveXrays(storage.data));
+    }
+  };
+
+  Object.defineProperty(window.wrappedJSObject, 'TestStorages', {
+    configurable: false,
+    writable: true,
```

Comment:
Now I see that I need it, as script injection happens in test ```setup``` method that redefines mock from the scratch for every test.